integration-tests/locales/all/src/test/java/io/quarkus/locales/it/LocalesIT.java

```diff
@@ -0,0 +1,114 @@
+package io.quarkus.locales.it;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import org.apache.http.HttpStatus;
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import io.restassured.RestAssured;
+
+/**
+ * A special case where we want to include all locales in our app.
+ * It must not matter which arbitrary locale we use, it must work here.
+ */
+@QuarkusIntegrationTest
+public class LocalesIT {
+
+    private static final Logger LOG = Logger.getLogger(LocalesIT.class);
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "en-US|en|United States",
+            "de-DE|de|Deutschland",
+            "de-AT|en|Austria",
+            "de-DE|en|Germany",
+            "zh-cmn-Hans-CN|cs|Čína",
+            "zh-Hant-TW|cs|Tchaj-wan",
+            "ja-JP-JP-#u-ca-japanese|sg|Zapöon"
+    })
+    public void testCorrectLocales(String countryLanguageTranslation) {
+        final String[] lct = countryLanguageTranslation.split("\\|");
+        LOG.infof("Triggering test: Country: %s, Language: %s, Translation: %s", lct[0], lct[1], lct[2]);
+        RestAssured.given().when()
+                .get(String.format("/locale/%s/%s", lct[0], lct[1]))
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(is(lct[2]))
+                .log().all();
+    }
```

Comment:
TIL that there is builtin support for this :)

```suggestion
    @ParameterizedTest
    @CsvSource({
            "en-US,en,United States",
            "de-DE,de,Deutschland",
            "de-AT,en,Austria",
            "de-DE,en,Germany",
            "zh-cmn-Hans-CN,cs,Čína",
            "zh-Hant-TW,cs,Tchaj-wan",
            "ja-JP-JP-#u-ca-japanese,sg,Zapöon"
    })
    public void testCorrectLocales(String country, String language, String translation) {
        LOG.infof("Triggering test: Country: %s, Language: %s, Translation: %s", country, language, translation);
        RestAssured.given().when()
                .get(String.format("/locale/%s/%s", country, language))
                .then()
                .statusCode(HttpStatus.SC_OK)
                .body(is(translation))
                .log().all();
    }
```